packages/flutter/lib/src/material/chip.dart

```diff
@@ -993,10 +993,13 @@ class _RawChipState extends State<RawChip> with MaterialStateMixin, TickerProvid
   OutlinedBorder _getShape(ThemeData theme, ChipThemeData chipTheme, ChipThemeData chipDefaults) {
     final BorderSide? resolvedSide = MaterialStateProperty.resolveAs<BorderSide?>(widget.side, materialStates)
       ?? MaterialStateProperty.resolveAs<BorderSide?>(chipTheme.side, materialStates)
+      ?? MaterialStateProperty.resolveAs<BorderSide?>(widget.shape?.side, materialStates)
```

Comment:
These two lines shouldn't be necessary. If the resolvedSide is null, then 
```dart
resolvedShape.copyWith(side: resolvedSide);
```
will not change the resolvedShape.

Reply:
This will work for Material 2 but not for Material 3. Material 3 has a default side property which overrides the default `shape.side`.

I just updated the default `shape.side` so we can use `resolvedShape.copyWith(side: resolvedSide);`

Please see the new commit https://github.com/flutter/flutter/pull/132941/commits/44dafce47c19b94e6c63145c33a3641a717893eb